### ISSUE TITLE
Doesn't work with Mail 10. 

### ISSUE THREAD

**Issue by serialmetrics** [nompute/GMailinator#43]
Does anyone have an updated version?

**Comment by jhollington** [comment]
Turns our Sierra adds a new info.plist key for the UUID, so you can make it load into Mail by adding the following section to _info.plist_:

```
    <key>Supported10.12PluginCompatibilityUUIDs</key>
    <array>
        <string>36CCB8BB-2207-455E-89BC-B9D6E47ABB5B</string>
    </array>

```

That said, while the plug-in seems to work for navigation, I'm getting exceptions on certain actions like delete and archive, so there's still something else in the code that needs to be updated for proper compatibility and I haven't had the time to dig further yet, but I'll post back if I figure anything out.

```
Exception Name: NSInvalidArgumentException
Description: -[MessageListViewController archiveMessages:]: unrecognized selector sent to instance 0x600000198390
User Info: (null)

```

**Comment by manlycode** [comment]
I've made a PR to automatically modify the plist to include the UUID of the building OS.

https://github.com/nompute/GMailinator/pull/44

**Comment by wunc** [comment]
Thanks. The new UUID was not that difficult to add, but the exception throws, particularly with the archive action (I use `e`), are definitely annoying. Would love it if someone could figure out a fix for those.

**Comment by acrollet** [comment]
I have a working version on Sierra here: https://github.com/Townk/GMailinator/pulls (including e for archive)

**Comment by serialmetrics** [comment]
Oh…. This works, splendidly! 

Thank you so much!!!!!!

Amir

**Comment by acrollet** [comment]
no problem! note for others, https://github.com/Townk/GMailinator is now sierra-compatible

**Comment by RaVbaker** [comment]
@acrollet Sorry, but this version is not working for me. I've latest Sierra 10.12.1 (16B2555). And it complains about plugin incompatibility. Could you provide a bundled version ? Because maybe it's because of some issues on my side when building in XCode. I got some warnings there and that's why I worry.

**Comment by acrollet** [comment]
@RaVbaker I haven't yet updated, probably there's a new UUID that needs to be added to the plist.

**Comment by RaVbaker** [comment]
Yes. I tried the uuid of my version of Mail.app but it didn't helped. So could you provide the bundle.zip file?

**Comment by earwiged** [comment]
Wondering if others have been able to get this running again after Sierra 10.1 (3251)?  I'm still getting plugin disabled errors.

**Comment by serialmetrics** [comment]
Hi @acrollet would you happen to have an updated version of Gmailinator for 10.12.2, that just came out today? :) Or, do you know what I need to edit to get it working?

**Comment by jgavris** [comment]
Working for me on 10.12.2 https://github.com/Townk/GMailinator/pull/3

**Comment by wunc** [comment]
@serialmetrics, if you want to do it manually, run the `find_uuid.sh` script in the project root and run it in Terminal, which should output the new Mail uuid. Then, open Xcode, find and edit the `info.plist` file, and add that new uuid to the `Supported10.12PluginCompatibilityUUIDs` key list. Save, then Build the project.
